x-pack/plugins/lists/server/saved_objects/migrations.ts

```diff
@@ -64,4 +67,25 @@ export const migrations = {
     },
     references: doc.references || [],
   }),
+  '7.12.0': (
+    doc: SavedObjectUnsanitizedDoc<ExceptionListSoSchema>
+  ): SavedObjectSanitizedDoc<ExceptionListSoSchema> => {
+    if (doc.attributes.list_id === ENDPOINT_TRUSTED_APPS_LIST_ID) {
+      return {
+        ...doc,
+        ...{
+          attributes: {
+            ...doc.attributes,
+            tags: [
+              ...(doc.attributes.tags || []),
+              ...(containsPolicyTags(doc.attributes.tags) ? [] : ['policy:all']),
+            ],
+          },
+        },
+        references: doc.references || [],
+      };
+    } else {
+      return { ...doc, references: doc.references || [] };
```

Comment:
Why is this needed? (same for above on line 85).

(FYI: you could also just mutate the object instead of "clone"'ing it

Reply:
it doesn't actually pass type check if I don't do this - apparently the input SO type has this as array | undefined while output type has it as array.